server/db-pg/seedHelpers.js

```diff
@@ -0,0 +1,182 @@
+  address: makeAddress(),
+  estDelivery: makeDeliveryTime(),
+  location: '"' + JSON.stringify(makeLocation()) + '"',
+  hours: '"' + JSON.stringify(makeHours()).replace(/"/g, '""') + '"',
+});
+
+const formatLine = object => Object.values(object).join(',');
+
+const clearCSV = () => {
+  const outputFile = `${__dirname}/restaurants.csv`;
+  fs.writeFileSync(outputFile, '');
+};
+
+const addToCSV = (items) => {
+  const lines = items.map(formatLine).join('\n') + '\n';
+
+  const outputFile = `${__dirname}/restaurants.csv`;
+  fs.appendFileSync(outputFile, lines, (err) => {
+    if (err) {
+      console.error(`Error appending to CSV: ${err}`);
+    }
+  });
+};
+
+const seedChunk = (start, end) => {
+  const restaurants = [];
+
+  for (let id = start; id <= end; id += 1) {
+    restaurants.push(makeRestaurant(id));
+  }
+
+  addToCSV(restaurants);
+};
+
+// Returns an array of sub-ranges that equally divide the given range. For
+// example, `makeChunkRanges(1, 100)` => [[1, 10], [11, 20], ..., [91, 100]]
+const makeChunkRanges = (start, end) => {
+  const numberOfChunks = 50; // Increase this number if you get out-of-memory errors when seeding
+  const rangeLength = end - start;
+  const chunkSize = Math.floor(rangeLength / numberOfChunks);
+
+  const ranges = [];
+  for (let i = start; i <= end; i += chunkSize + 1) {
+    ranges.push([i, Math.min(i + chunkSize, end)]);
+  }
+
+  return ranges;
+};
+
+const seedInChunks = () => {
+  const chunkRanges = makeChunkRanges(startId, endId);
+  console.log(`Generating ${chunkRanges.length} chunks...`);
+  clearCSV();
+
+  chunkRanges.forEach((range, i) => {
+    process.stdout.write((i + 1).toString());
+    seedChunk(...range);
+    process.stdout.write('.');
+  });
+
+  console.log('\nAll chunks generated.');
+};
+
+const loadCSVIntoDatabase = () => new Promise((resolve, reject) => {
+  const command = exec(`psql -d ${database} -c "copy restaurants from '${__dirname}/restaurants.csv' csv delimiter ','"`);
+
+  command.stderr.on('data', (err) => {
+    reject(err);
+  });
+
+  command.on('close', (code) => {
+    if (code === 0) {
+      resolve('CSV data loaded into database.');
+    } else {
+      reject(new Error('Unable to add CSV data into database'));
+    }
+  });
+});
+
+const clearTable = () => new Promise((resolve, reject) => {
+  const command = exec(`psql -d ${database} -f ${__dirname}/schema.sql`);
+
+  command.on('close', (code) => {
+    if (code === 0) {
+      resolve('`restaurants` table cleared.');
+    } else {
+      reject(new Error('Unable to clear table'));
+    }
+  });
+});
+
+const handleSeeding = () => {
+  console.log(`Seeding database with ${endId - startId + 1} items...`);
+
+  return clearTable()
+    .then(() => seedInChunks())
+    .then(() => console.log('Loading CSV into database...'))
+    .then(() => loadCSVIntoDatabase())
+    .then(() => {
+      console.log('CSV loaded into database');
+      console.log('Database seeded successfully. Have a nice day.');
```

Comment:
Nice touch here, this all looks great to me. Congrats on figuring out the seeding!